client/push/push_manager.ts

```diff
@@ -19,25 +19,46 @@ declare var navigator: any;
 export default class PushManager {
   private isRegistered_: boolean = false;
 
-  constructor() {
+  constructor() {}
+
+  async isRegistered(): Promise<boolean> {
+    // Not implemented yet
+    return false;
+  }
+
+  async register(key: string): Promise<boolean> {
+    // Not implemented yet
     if (navigator.serviceWorker) {
       navigator.serviceWorker.register('/push_service_worker.js')
         .then((registration: ServiceWorkerRegistration) => {
-          this.isRegistered_ = true;
+
+          if (!registration.pushManager) {
+            return false;
+          }
+
+          registration.pushManager.subscribe({
+            userVisibleOnly: true
```

Comment:
We might need to take a server key?

Reply:
Yes, we need it. 
Should I use the server key from the link below?https://github.com/haeungun/absolute/blob/06d7b71bdcfee70911d065d7ec024d5275ebe2fa/client/push/push_key.js

Reply:
If we use vapid, yes and it need converting to uint8array.
If we dont use vapid, it doesnt need. (Should use sender_id of manifest.json from firebase in case of gcm)
Recommand to use vapid